miden-crypto/src/hash/algebraic_sponge/mod.rs

```diff
@@ -49,40 +55,47 @@ const BINARY_CHUNK_SIZE: usize = 7;
 ///
 /// The constants are defined for tests only because the exponentiations in the code are unrolled
 /// for efficiency reasons.
-#[cfg(test)]
-const ALPHA: u64 = 7;
-#[cfg(test)]
-const INV_ALPHA: u64 = 10540996611094048183;
+#[cfg(all(test, feature = "std"))]
+pub(crate) const ALPHA: u64 = 7;
+#[cfg(all(test, feature = "std"))]
+pub(crate) const INV_ALPHA: u64 = 10540996611094048183;
 
 // ALGEBRAIC SPONGE
 // ================================================================================================
 
-pub(crate) trait AlgebraicSponge {
+// Note: This trait must remain `pub` (not `pub(crate)`) because it's re-exported
+// at the crate root (lib.rs) as part of the public API.
+pub trait AlgebraicSponge {
```

Comment:
Same question as in one of the other comments: why does this need to be a part of the public API?